mage_ai/orchestration/pipeline_scheduler.py

```diff
@@ -1520,23 +1598,59 @@ def schedule_with_event(event: Dict = None):
         if sync_config and sync_config.sync_on_pipeline_run:
             git_sync_result = run_git_sync(lock=lock, sync_config=sync_config)
 
-        for p in matched_pipeline_schedules:
-            payload = dict(
-                execution_date=datetime.now(tz=pytz.UTC),
-                pipeline_schedule_id=p.id,
-                pipeline_uuid=p.pipeline_uuid,
-                variables=merge_dict(p.variables or dict(), dict(event=event)),
-            )
-            pipeline_run = PipelineRun.create(**payload)
-            pipeline_scheduler = PipelineScheduler(pipeline_run)
+        matched_pipeline_uuids = list(set([s.pipeline_uuid for s in matched_pipeline_schedules]))
```

Comment:
Same here for event trigger. This method just needs to create the pipeline run.
The scheduling or cancellation logic can be put in `schedule_all` method.